src/backend/src/migrate.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    mutate_state,
+    types::{Candid, StoredPrincipal},
+};
+use candid::{decode_one, CandidType, Principal};
+use serde::Deserialize;
+use shared::types::{
+    custom_token::CustomToken, token::UserToken, user_profile::StoredUserProfile, Timestamp,
+};
+
+/// A chunk of data to be migrated.
+///
+/// Note: Given that the migration moves data types that may be private, data is transferred with candid type `Vec<u8>`
```

Comment:
Have you considered using `cbor` instead? Cbor is much more efficient and given this is about bulk data, it might make a difference.

Reply:
The performance now is even worse than that:  When the data is read from the `BTreeMap` it is parsed from candid bytes into Rust structs and there is a corresponding serialization when the data is re-inserted.  I thought about getting the raw bytes from the `BTreeValue` and transmitting those; that would be a significant improvement.  But the bulk data is still quite small and downtime is not a priority, so I decided not to optimize.  Using CBOR is a good idea but in a similar way this type of optimization is not a priority.  I hope that makes sense :-)